eng/Build.ps1

```diff
@@ -437,6 +437,11 @@ try {
     $buildTool = InitializeBuildTool
     $toolsetBuildProj = InitializeToolset
     TryDownloadDotnetFrameworkSdk
+
```

Comment:
@brettfo , this was the main problem, the ci sets DOTNET_ROOT to a the global location.   Here we set it to the local download location first, so that the local and global frameworks are selectable by dotnet.exe